Assets/HoloToolkit/Utilities/Scripts/Solvers/SolverConstantViewSize.cs

```diff
@@ -0,0 +1,184 @@
+		{
+			get { return objectScalePercent; }
+		}
+
+		// 0 to 1 between MinDistance and MaxDistance.  If current < max, object is potentially on a surface [or some other condition like interpolating] (since it may still be on surface, but scale percent may be clamped at max)
+		// This value is subject to inaccuracies due to smoothing/interpolation/momentum
+		public float CurrentDistancePercent
+		{
+			get { return objectDistancePercent; }
+		}
+		#endregion
+
+		#region private members
+		private float fovScalar = 1f;
+		private float objectSize = 1f;
+		private float objectScalePercent = 1f;
+		private float objectDistancePercent = 1f;
+		#endregion
+
+		protected override void Start()
+		{
+            base.Start();
+
+            float baseSize = CalculateObjectSize();
+
+			if (baseSize > 0)
+			{
+				objectSize = baseSize;
+			}
+			else
+			{
+				Debug.LogError("ConstantViewSize: Object base size calculate was 0, defaulting to 1");
+				objectSize = 1f;
+			}
+		}
+
+		/// <summary>
+		///   Attempts to calculate the size of the bounds which contains all child renderers.
+		///   This may be tricky to use, as this happens during initialization, while the app may
+		///   be undergoing scaling by other solvers/components.  Thus, the size calculation might
+		///   be inaccurate.  It's probably a better idea to use ManualObjectSize just to be sure.
+		/// </summary>
+		/// <returns> Object diameter </returns>
+		private float CalculateObjectSize()
+		{
+			if (ManualObjectSize > 0)
+			{
+				return ManualObjectSize;
+			}
+
+			Vector3 rootScale = transform.root.localScale;
+			transform.root.localScale = Vector3.one;
+
+			float maxSize = 1f;
+
+			Bounds combinedBounds = new Bounds(transform.position, Vector3.zero);
+			Renderer[] renderers = this.GetComponentsInChildren<Renderer>();
+
+			foreach (Renderer render in renderers)
+			{
+				combinedBounds.Encapsulate(render.bounds);
+			}
+
+			maxSize = combinedBounds.extents.magnitude;
+
+			transform.root.localScale = rootScale;
+
+			return maxSize;
+		}
+
+
+		public override void SolverUpdate()
+		{
+			float lastScalePct = objectScalePercent;
+			AdjustSizeForView(solverHandler.TransformTarget);
+			float scaleDiff = (objectScalePercent - lastScalePct) / solverHandler.DeltaTime;
+
+			if (scaleDiff > ScaleBuffer)
+			{
+				ScaleState = ScaleStateEnum.Growing;
+			}
+			else if (scaleDiff < -ScaleBuffer)
+			{
+				ScaleState = ScaleStateEnum.Shrinking;
+			}
+			else
+			{
+				ScaleState = ScaleStateEnum.Static;
+			}
+		}
+
+		/// <summary>
+		///   Returns the scale to be applied based on the FOV.  This scale will be multiplied by distance as part of
+		///   the final scale calculation, so this is the ratio of vertical fov to distance.
+		/// </summary>
+		/// <returns> Scale of vFOV </returns>
+		public float GetFOVScalar()
+		{
+            float camFOVrad = (Camera.main.aspect * Camera.main.fieldOfView) * Mathf.Deg2Rad;
+
+			float sinfov = Mathf.Sin(camFOVrad / 2f);
```

Comment:
nit: * 0.5f

Reply:
Whats the reasoning here for changing Mathf.Sin(camFOVrad / 2f); to *0.5f?

Reply:
Multiplication is faster than division